### DISCUSSION TITLE
DC Voltage sensor always 0.

### DISCUSSION THREAD

**Discussion by Claudio1L** [alexdelprete/ha-abb-powerone-pvi-sunspec#64] [Support]
Ciao ALex

Don't know if it's a my issue. From the first installation I always had 0 for the -DC voltage. Watt and Current are fine, AC is fine as well
I could create a template to derive voltage from power and current, but just curious if can be fixed.

![immagine](https://user-images.githubusercontent.com/97904190/230709677-6df51d45-b5a1-45f6-ad08-123e2cb7b904.png)

thanks as always

**Comment by Claudio1L** [comment]
The maximum would be having the two string values :-). Would be useful to see if there are issues on one string

![immagine](https://user-images.githubusercontent.com/97904190/230709965-ad59b0f8-8971-43bf-8d2f-fd5f5aab4867.png)

**Comment by alexdelprete** [comment]
I have a three-phase inverter with two strings, and it works fine here. Tell me about your system, is it three-phase or monophase? you have two strings?

![image](https://user-images.githubusercontent.com/7027842/230722874-c9584c9a-52c7-4dd6-912e-bd95d6e83e49.png)

**Comment by Claudio1L** [comment]
it's a monophase,  ABB/Fimer UNO-DM-4.6-TL-PLUS with integrated wifi
thanks

**Comment by alexdelprete** [comment]
How many strings?

**Comment by Claudio1L** [comment]
two strings, the ones you see in the previous attach

**Comment by Claudio1L** [comment]
I see everything aggregated

**Comment by alexdelprete** [comment]
There's no aggregation when reading the strings.

Please show me a screenshot with ALL the sensors.

**Comment by Claudio1L** [comment]
![immagine](https://user-images.githubusercontent.com/97904190/230723645-229eda2e-21fc-4202-b5fd-b1a16bc678af.png)

**Comment by alexdelprete** [comment]
Use qmodmaster and read the registers, like in this screenshot, so we can check the raw data: at address 70, you should have value 101 (monophase inverter) and the DC Voltage is at address 99.

![image](https://user-images.githubusercontent.com/7027842/230727966-b99a9cc6-f264-4d4b-8ba9-903ba6bfdca3.png)

**Reply by alexdelprete** [reply]
Also from start address 120 to check the strings data:

![image](https://user-images.githubusercontent.com/7027842/230728725-b37c7281-9f43-499b-8ecc-15b0d178b497.png)

**Comment by Claudio1L** [comment]
not sure I got the right registers. Starting at 0 was reading nothing
![Pasted Graphic 1](https://user-images.githubusercontent.com/97904190/230739739-15071fb4-706e-4509-b2f5-0a5c9ad18df7.png)
![immagine](https://user-images.githubusercontent.com/97904190/230739773-840c6ca4-e0fe-4b7b-a441-3ab7e40234e3.png)

**Reply by alexdelprete** [reply]
As you can see in the following image, I highlighted addresses 97 (DC Amps), 99 (DC Voltage), 101 (DC Watts).

![image](https://user-images.githubusercontent.com/7027842/230746089-45eaa26e-0eca-4f88-ab96-0cf66f7b3ec3.png)

The DC Voltage register has a wrong value (65535). That's why you see a wrong value in the sensor.

The register map is an excel file that FIMER sent me when I started developing the component. It is the register map for the VSN300/VSN700 wifi dataloggers. Here's the content for those 3 registers:

![image](https://user-images.githubusercontent.com/7027842/230746208-406fbd68-70ef-4b16-b892-d642c5a1beda.png)

So, I can't tell you why the value is reported wrongly, maybe it's related to the firmware or something else, but there's not much I can do about it.

**Reply by alexdelprete** [reply]
The other more serious issue is the fact that the strings were not read in your case, and I found out why.

Following the register map excel file that FIMER sent me, I know that the strings data is available from address 122, where I should have a value of 160 (that's the SunSpec Model 160 for MPTT extensions).

![image](https://user-images.githubusercontent.com/7027842/230746332-c7b1237c-8860-4e79-9eb7-76ed8cf956e1.png)

But in your case, at address 122 we have a value of 120 which is not expected:

![image](https://user-images.githubusercontent.com/7027842/230746379-3817733e-96e6-4acc-97fb-271efc469c32.png)

So I really don't know what I can do, other than suggesting to contact FIMER and ask for the register map for your specific model, and maybe also ask them why the VSN300 SunSpec Register Map does not match data coming from your inverter. Maybe they changed the map in recent firmware, but I don't really know.

**Comment by alexdelprete** [comment]
One thing to try: uninstall my integration, and try [ha-sunspec](https://github.com/CJNE/ha-sunspec) integration, available in HACS.

`ha-sunspec` is based on sunspec official libraries. if you can test it and send me the screenshot of all the reported sensors we can double-check if there's some difference.

**Comment by Claudio1L** [comment]
ok, I did it.

configuration:
![immagine](https://user-images.githubusercontent.com/97904190/230756724-78d46e01-39ac-40a5-ab41-36a1c71ea286.png)

![immagine](https://user-images.githubusercontent.com/97904190/230756745-38b128f8-1ac5-4505-8501-1ab1c4a025ea.png)

he found 2 devices

![immagine](https://user-images.githubusercontent.com/97904190/230756803-51d7b231-333f-414f-a86e-c4b445f39a9b.png)

![immagine](https://user-images.githubusercontent.com/97904190/230756824-efdd35a8-0852-43a0-bc9a-f9380ffa88ba.png)

![immagine](https://user-images.githubusercontent.com/97904190/230756849-b1466470-35e0-45a9-901c-5fc1505374aa.png)

![immagine](https://user-images.githubusercontent.com/97904190/230756872-99cbf4d7-a335-4c48-bff8-b28b294f83d1.png)


The extension model seems to have string values

![immagine](https://user-images.githubusercontent.com/97904190/230756941-38ca109e-db73-4819-a178-6606ecc2b5fb.png)

**Comment by Claudio1L** [comment]
ok, it seems to have the missing info, probably is reading them from other registers. Now, Alex I could also use this integration if requires too much time modify yours or if you are not interested to modify it. I understand for sure

**Reply by alexdelprete** [reply]
it's easy to modify, but I need to have the updated register map and check what has changed.

I noticed other 2-3 users have the same issue, but all the rest don't, so it's not easy to determine how to change if it's only for specific models, or, versions of firmware.

What datalogger does your inverter use? is there the model in the webpage?

You don't need to ask me what integration to use, you use what's best for you, that's why I recommended an alternative. :)

**Comment by Claudio1L** [comment]
At the end, I decided to maintain your integration for the AC part, and the sunspace integration only for the DC part.

This because of different behaviour of the sunspace integration from your.

Your integration, maintain the entities value when the inverter shuts down in the night, and it’s fine for my statistics reports. The only exception if I reboot HA. The sunspec, instead, provides “unknown” value as soon as the inverter shutdown.
For me this is an issue because it broke all my counters.
